credentials/credentials.go

```diff
@@ -50,6 +51,49 @@ type PerRPCCredentials interface {
 	RequireTransportSecurity() bool
 }
 
+// SecurityLevel defines the protection level on an established channel.
+type SecurityLevel int
+
+const (
+	// NoSecurity indicates a channel is a plaintext channel that does not provide any protection.
+	NoSecurity SecurityLevel = iota
+	// IntegrityOnly indicates a channel only provides integrity protection.
+	IntegrityOnly
+	// PrivacyAndIntegrity indicates a channel provides both privacy and integrity protection.
+	PrivacyAndIntegrity
+)
+
+// String returns SecurityLevel in a string format.
+func (s SecurityLevel) String() string {
+	switch s {
+	case NoSecurity:
+		return "NoSecurity"
+	case IntegrityOnly:
+		return "IntegrityOnly"
+	case PrivacyAndIntegrity:
+		return "PrivacyAndIntegrity"
+	}
+	return ""
+}
+
+// CommonAuthInfo contains authenticated information common to AuthInfo implementations.
```

Comment:
... It should be embedded in a struct implementing `AuthInfo` to provide additional information about the credentials.

We should add a comment on `AuthInfo` and also `ClientHandshake` and `ServerHandshake` in `TransportCredentials`.

Reply:
Done in the next commit.